analyses/snv-callers/compare_snv_callers_plots.Rmd

```diff
@@ -0,0 +1,284 @@
+```
+
+Set up output directories. 
+
+```{r}
+base_results_dir <- "results"
+base_plots_dir <- "plots"
+```
+
+Make new directories for the comparison analysis.
+
+```{r}
+consensus_results_dir <- file.path(base_results_dir, "consensus")
+consensus_plots_dir <- file.path(base_plots_dir, "consensus")
+
+# Make caller specific plots folder
+if (!dir.exists(consensus_results_dir)) {
+  dir.create(consensus_results_dir)
+}
+# Make caller specific plots folder
+if (!dir.exists(consensus_plots_dir)) {
+  dir.create(consensus_plots_dir)
+}
+```
+
+## Import data 
+
+Load in the merged data.frames from `03-merge_callers.R` output. 
+
+```{r}
+# Master VAF data.frame
+vaf_df <- readr::read_rds(file.path(consensus_results_dir, 
+                                    "all_callers_vaf.rds"))
+
+# Master TMB data.frame
+tmb_df <- readr::read_rds(file.path(consensus_results_dir, 
+                                    "all_callers_tmb.rds"))
+
+# Full mutation list
+mutation_id_list <- readr::read_rds(file.path(consensus_results_dir, 
+                                              "mutation_id_list.rds"))
+
+# Callers per mutation data.frame
+callers_per_mutation <- readr::read_rds(file.path(consensus_results_dir, 
+                                                  "callers_per_mutation.rds"))
+```
+
+## Upset graph
+
+Make the upset plot. 
+
+```{r}
+# Make the plot
+upset <- UpSetR::upset(UpSetR::fromList(mutation_id_list),
+  order.by = "freq",
+  text.scale = 2
+)
+
+# Save this plot
+png(file.path(consensus_plots_dir, "upset_plot.png"), width = 900, height = 900)
+upset
+dev.off()
+
+# Print this out
+upset
+```
+
+## VAF for combination sets of callers
+
+Graph mutations by combinations of callers and plot the VAF density.
+Get list of callers per mutation and the median vaf for each. 
+Graph the median VAF for each combination of callers.
+
+```{r}
+# Make this plot
+callers_per_mutation %>%
+  ggplot2::ggplot(ggplot2::aes(x = caller_combo, y = median_vaf)) +
+  ggplot2::geom_violin() +
+  ggplot2::theme_classic() +
+  ggupset::scale_x_mergelist(sep = "-") +
+  ggupset::axis_combmatrix(sep = "-") +
+  ggplot2::xlab("") +
+  ggplot2::ylab("Median VAF Across Callers")
+```
+
+```{r}
+# Save this plot
+ggplot2::ggsave(file.path(consensus_plots_dir, "upset_median_vaf_plot.png"))
+```
+
+## Base Changes 
+
+Set up the summarized form of the base change information. 
+
+```{r}
+# Summarize by the number of times each base change shows up in each category.
+perc_change_df <- vaf_df %>%
+  dplyr::count(caller, change, name = "count") %>%
+  dplyr::add_count(caller, wt = count) %>%
+  dplyr::mutate(percent = count / n) %>%
+  dplyr::filter(!grepl("G>G|A>A|N>A|N>T", change))
+
+# Need to rearrange the order of the base change categories
+perc_change_df <- perc_change_df %>%
+  dplyr::mutate(
+    change = as.factor(change),
+    change = factor(change, c(grep("ins|del|long_change", levels(change), value = TRUE, invert = TRUE), "ins", "del", "long_change"))
```

Comment:
Break up this line? The indentation here (or lack thereof) looks a little off.

Reply:
This line is crazy. Agreed. Will try to neaten it up.

Reply:
didn't I suggest forcats::fct_relevel() for this in a different place?